common/lib/xmodule/xmodule/fields.py

```diff
@@ -252,3 +253,37 @@ def enforce_type(self, value):
             return value
 
         return self.from_json(value)
+
+
+class ScoreField(JSONField):
+    """
+    Field for blocks that need to store a Score. XBlocks that implement
+    the ScorableXBlockMixin may need to store their score separately
+    from their problem state, specifically for use in staff override
+    of problem scores.
+    """
+    MUTABLE = False
+
+    def from_json(self, value):
+        if value is None:
+            return value
+        if isinstance(value, Score):
+            return value
+
+        if ('raw_earned' not in value or 'raw_possible' not in value) or any(
+            key not in ['raw_earned', 'raw_possible'] for key in value.keys()
+        ):
+            raise TypeError('Scores must contain only a raw earned and raw possible value.')
+
+        raw_earned = value['raw_earned']
+        raw_possible = value['raw_possible']
+
+        if raw_possible < 0:
+            raise ValueError('raw_possible must be a positive number.')
+
+        if not (0 <= raw_earned <= raw_possible):
+            raise ValueError('raw_earned must be between zero and raw_possible.')
```

Comment:
For all of these `TypeError` and `ValueError` instances, please keep in mind that at some point, somebody is going to read them in a log file or a console. This is a pretty low level operation, and that person is going to want additional context to figure out what's going on. So for example, say you're doing something completely unrelated to grading and suddenly see this in a console:

```python
TypeError('Scores must contain only a raw earned and raw possible value.')
```
What are the questions you'd ask at that point? What's the name of the field (`self.display_name`)? What are the exact names of the fields it was expecting (`raw_earned`, `raw_possible`)? What did it get instead (`value.keys()`)? Was it a silly typo somewhere, or are we passing an entirely inappropriate data structure somehow?

The same thing goes for the `ValueError` cases. If `raw_possible` is negative, what is that number? If `raw_earned` doesn't satisfy the range condition, what were the values being compared? You might also want to use the `repr` versions of these vars, in case there are weird places where strings are being passed instead of numbers (it's happened before -- the joys of dynamic typing). Note: Python 2.7 will allow comparisons between numbers and strings, they just won't do what most people would expect.

Please flesh out these exception messages a bit more to add that additional debugging context.